### PR TITLE
feat(releaseci): adding a dedicated disk/pv/pvc for release.ci as standard-zrs

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/4044

preparing the migration from premium to standard ZRS

### CODE REVIEW

File: locals.tf
Comment (dduportal) [inline locals.tf:68]: As checked together, you might want to update the PR here to apply the same pattern as in https://github.com/jenkins-infra/azure/pull/769 to avoid local and make the PVC -> PV -> Disk dependency chain explicit

File: privatek8s.tf
Comment (dduportal) [inline privatek8s.tf:329]:
```suggestion
  provider               = kubernetes.privatek8s
```

File: release.ci.jenkins.io.tf
Comment (dduportal) [inline release.ci.jenkins.io.tf:3]:
```suggestion
  location = var.location
```

Ref. https://github.com/jenkins-infra/azure/blob/main/variables.tf#L1

File: release.ci.jenkins.io.tf
Comment (dduportal) [inline release.ci.jenkins.io.tf:19]:
```suggestion
  provider = kubernetes.privatek8s
```

File: release.ci.jenkins.io.tf
Comment (dduportal) [inline release.ci.jenkins.io.tf:40]:
```suggestion
  provider = kubernetes.privatek8s
```

File: release.ci.jenkins.io.tf
Comment (dduportal) [inline release.ci.jenkins.io.tf:69]:
```suggestion
  principal_id       = azurerm_kubernetes_cluster.privatek8s.identity[0].principal_id
```

**Review by dduportal — CHANGES_REQUESTED** [review]
First pass of review:

- Overall work looks good 👍 
- Blockers (see suggestions) due to copy and paste

**Comment by smerle33** [comment]
> First pass of review:
> 
>     * Overall work looks good 👍
> 
>     * Blockers (see suggestions) due to copy and paste

can't believed I missed it

**Review by dduportal — APPROVED** [review]
`Plan: 7 to add, 0 to change, 0 to destroy.`

LGTM

Please wait for the end of the weekly release 2.465 (today) before merging this PR

**Pull request merged**
